hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1152,20 +1155,49 @@ func (r *NodePoolReconciler) listSecrets(nodePool *hyperv1.NodePool) ([]corev1.S
 	return filtered, nil
 }
 
-func (r *NodePoolReconciler) listAWSMachineTemplates(nodePool *hyperv1.NodePool) ([]capiaws.AWSMachineTemplate, error) {
-	awsMachineTemplateList := &capiaws.AWSMachineTemplateList{}
-	if err := r.List(context.Background(), awsMachineTemplateList); err != nil {
+func (r *NodePoolReconciler) listMachineTemplates(nodePool *hyperv1.NodePool) ([]client.Object, error) {
```

Comment:
can we have a unit test for this func that validates it list a precreated template for a given platform?

Reply:
Added a UT